File: android/versioned-abis/expoview-abi39_0_0/src/main/java/abi39_0_0/host/exp/exponent/modules/universal/sensors/ScopedMagnetometerService.java
package abi39_0_0.host.exp.exponent.modules.universal.sensors;

import java.util.Collections;
import java.util.List;

import abi39_0_0.org.unimodules.core.interfaces.InternalModule;
import abi39_0_0.org.unimodules.interfaces.sensors.services.MagnetometerService;
import host.exp.exponent.kernel.ExperienceKey;
import host.exp.exponent.kernel.services.sensors.SubscribableSensorKernelService;

public class ScopedMagnetometerService extends BaseSensorService implements InternalModule, MagnetometerService {
  public ScopedMagnetometerService(ExperienceKey experienceKey) {
    super(experienceKey);
  }

  @Override
  protected SubscribableSensorKernelService getSensorKernelService() {
    return getKernelServiceRegistry().getMagnetometerKernelService();
  }

  @Override
  public List<Class> getExportedInterfaces() {
    return Collections.<Class>singletonList(MagnetometerService.class);
  }
}

File: android/versioned-abis/expoview-abi48_0_0/src/main/java/abi48_0_0/host/exp/exponent/modules/universal/sensors/ScopedMagnetometerService.kt
package abi48_0_0.host.exp.exponent.modules.universal.sensors

import abi48_0_0.expo.modules.core.interfaces.InternalModule
import abi48_0_0.expo.modules.interfaces.sensors.services.MagnetometerServiceInterface
import host.exp.exponent.kernel.ExperienceKey
import host.exp.exponent.kernel.services.sensors.SubscribableSensorKernelService

class ScopedMagnetometerService(experienceKey: ExperienceKey) : BaseSensorService(experienceKey), InternalModule, MagnetometerServiceInterface {
  override val sensorKernelService: SubscribableSensorKernelService
    get() = kernelServiceRegistry.magnetometerKernelService

  override fun getExportedInterfaces(): List<Class<*>> {
    return listOf(MagnetometerServiceInterface::class.java)
  }
}
